File: Balanced_Parenthesis.java
import java.util.*;
class Driver{
    public static void main(String[] args){
        Map<Character,Integer> mpp=new HashMap<Character,Integer>();
        mpp.put('<',1);
        mpp.put('(',3);
        mpp.put('{',5);
        mpp.put('[',7);
        mpp.put('>',2);
        mpp.put(')',6);
        mpp.put('}',10);
        mpp.put(']',14);
        Stack<Character> stk=new Stack<Character>();
        Scanner sc=new Scanner(System.in);
        boolean flag=true;
        System.out.println("Enter a string");
        String str=sc.next();
        for(int i=0;i<str.length();i++){
            if(str.charAt(i)=='<' || str.charAt(i)=='(' || str.charAt(i)=='{'|| str.charAt(i)=='[')
                stk.push(str.charAt(i));
            else if(stk.empty()){
                flag=false;
                break;
            }
            else if(mpp.get(str.charAt(i))==2*mpp.get(stk.peek())){
                stk.pop();
            }
        }
        if(!stk.empty()) flag=false;
        if(flag)
            System.out.println("Balanced Parenthesis");
        else
            System.out.println("Not Balanced Parenthesis");
    }
}

File: Balanced_Parenthesis.cpp
#include<iostream>
#include<stack>
#include<map>
using namespace std;
int main(){
    string str;
    map<char,int> mpp;
    mpp.insert({'<',1});
    mpp.insert({'(',3});
    mpp.insert({'[',5});
    mpp.insert({'{',7});
    mpp.insert({'>',2});
    mpp.insert({')',6});
    mpp.insert({']',10});
    mpp.insert({'}',14});
    bool flag=true;
    stack<char>stk;
    cout<<"Enter a string\n";
    cin>>str;
    for(int i=0;i<str.length();i++){
        if(str[i]=='<' || str[i]=='(' || str[i]=='{'|| str[i]=='[')
            stk.push(str[i]);
        else if(stk.empty()){
            flag=false;
            break;
        }
        else if(mpp[str[i]]==2*mpp[stk.top()]){
            stk.pop();
        }
    }
    if(!stk.empty())flag=false;
    flag?cout<<"Balanced Parenthesis\n":cout<<"Not a Balanced Parenthesis\n";
}